hazelcast/src/main/java/com/hazelcast/config/WanQueueFullBehavior.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+/**
+ * Determines the behavior of WAN replication impl. in case of WAN event queues are full
+ */
+public enum WanQueueFullBehavior {
+
+    /**
+     * Instruct WAN repl. impl to drop new events when WAN event queues are full
+     */
+    DISCARD_AFTER_MUTATION(0),
```

Comment:
Actually this is the existing behavior of WAN replication and will be default behavior of the new version also.
When `DISCARD_AFTER_MUTATION` is chosen, mutating operation (put/remove/etc...) will be applied but WAN event will be discarded if WAN replication queue is full.

When `THROW_EXCEPTION` is chosen, a queue size check will be performed in `beforeRun` of mutating operations and `WANReplicationQueueFullException` will be throwed if WAN replication queue is full, therefore mutating operation (put/remove/etc...) won't run.